hooks/pre/40-el8_modules.rb

```diff
@@ -0,0 +1,8 @@
+if (module_enabled?('katello') || module_enabled?('foreman_proxy_content')) && el8?
+  execute!('dnf -y module enable pki-core')
+
+  if local_postgresql?
+    execute!('dnf -y module reset postgresql', false, true)
+    execute!('dnf -y module enable postgresql:12', false, true)
```

Comment:
I think this fails if some postgres package was already installed. I think the katello meta package pulls that in, so I doubt this really works.

If the workflow is:
* install foreman-installer-katello
* run foreman-installer
* let foreman-installer install katello/foreman-proxy-content
Then it will probably work.

Reply:
From my testing, and I can do a little more. If you already have say postgresql 10 installed from that module, the reset clears the dnf metadata that prevents update. 

```
[root@42c3f74ab611 /]# rpm -q postgresql
postgresql-10.15-1.module_el8.3.0+619+dbc95fbc.x86_64
[root@42c3f74ab611 /]# ls /var/lib/               
alternatives/ dbus/         dhclient/     dnf/          games/        initramfs/    misc/         portables/    private/      rpm/          rpm-state/    selinux/      systemd/      
[root@42c3f74ab611 /]# rpm -q postgresql
postgresql-10.15-1.module_el8.3.0+619+dbc95fbc.x86_64
[root@42c3f74ab611 /]# dnf module reset postgresql
Failed to set locale, defaulting to C.UTF-8
Last metadata expiration check: 0:00:51 ago on Mon Feb 15 14:14:44 2021.
Dependencies resolved.
==========================================================================================================================================================================================================================================================================================
 Package                                                              Architecture                                                        Version                                                              Repository                                                            Size
==========================================================================================================================================================================================================================================================================================
Resetting modules:
 postgresql                                                                                                                                                                                                                                                                              

Transaction Summary
==========================================================================================================================================================================================================================================================================================

Is this ok [y/N]: y 
Complete!
[root@42c3f74ab611 /]# dnf module enable postgresql:12
Failed to set locale, defaulting to C.UTF-8
Last metadata expiration check: 0:00:57 ago on Mon Feb 15 14:14:44 2021.
Dependencies resolved.
==========================================================================================================================================================================================================================================================================================
 Package                                                              Architecture                                                        Version                                                              Repository                                                            Size
==========================================================================================================================================================================================================================================================================================
Enabling module streams:
 postgresql                                                                                                                               12                                                                                                                                             

Transaction Summary
==========================================================================================================================================================================================================================================================================================

Is this ok [y/N]: y
Complete!
[root@42c3f74ab611 /]# dnf install postgresql
Failed to set locale, defaulting to C.UTF-8
Last metadata expiration check: 0:01:03 ago on Mon Feb 15 14:14:44 2021.
Package postgresql-10.15-1.module_el8.3.0+619+dbc95fbc.x86_64 is already installed.
Dependencies resolved.
==========================================================================================================================================================================================================================================================================================
 Package                                                         Architecture                                                Version                                                                                 Repository                                                      Size
==========================================================================================================================================================================================================================================================================================
Upgrading:
 postgresql                                                      x86_64                                                      12.5-1.module_el8.3.0+620+f1abd701                                                      AppStream                                                      1.5 M

Transaction Summary
==========================================================================================================================================================================================================================================================================================
Upgrade  1 Package

Total download size: 1.5 M
Is this ok [y/N]: y
Downloading Packages:
postgresql-12.5-1.module_el8.3.0+620+f1abd701.x86_64.rpm                                                                                                                                                                                                  4.3 MB/s | 1.5 MB     00:00    
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Total                                                                                                                                                                                                                                                     3.1 MB/s | 1.5 MB     00:00     
Running transaction check
Transaction check succeeded.
Running transaction test
Transaction test succeeded.
Running transaction
  Preparing        :                                                                                                                                                                                                                                                                  1/1 
  Upgrading        : postgresql-12.5-1.module_el8.3.0+620+f1abd701.x86_64                                                                                                                                                                                                             1/2 
  Cleanup          : postgresql-10.15-1.module_el8.3.0+619+dbc95fbc.x86_64                                                                                                                                                                                                            2/2 
  Running scriptlet: postgresql-10.15-1.module_el8.3.0+619+dbc95fbc.x86_64                                                                                                                                                                                                            2/2 
  Verifying        : postgresql-12.5-1.module_el8.3.0+620+f1abd701.x86_64                                                                                                                                                                                                             1/2 
  Verifying        : postgresql-10.15-1.module_el8.3.0+619+dbc95fbc.x86_64                                                                                                                                                                                                            2/2 

Upgraded:
  postgresql-12.5-1.module_el8.3.0+620+f1abd701.x86_64                                                                                                                                                                                                                                    

Complete!
[root@42c3f74ab611 /]# 
```

Reply:
It looks like `rubygem-pg` depends on `libpq` which is **not** in the module. That means it's not version dependent, which is good for us.

It also appears nothing pulls in `psql` (which comes from the postgresql) pacakge even though `katello-debug` uses it. (Also triggered me to open https://github.com/theforeman/foreman-packaging/pull/6315).

That means it may work.

The only thing that concerns me is idempotency. On the next run (when we have installed postgresql), it will also reset and enable it again.